build.psm1

```diff
@@ -286,122 +287,191 @@ function Start-PSBuild {
-            Start-NativeExecution { cmake -DCMAKE_BUILD_TYPE=Debug . }
-            Start-NativeExecution { make -j }
-            Start-NativeExecution { ctest --verbose }
-        } finally {
-            Pop-Location
-        }
+            try {
+                Push-Location $Native
+                Start-NativeExecution { cmake -DCMAKE_BUILD_TYPE=Debug . }
+                Start-NativeExecution { make -j }
+                Start-NativeExecution { ctest --verbose }
+            } finally {
+                Pop-Location
+            }
 
-        if (-not (Test-Path $Lib)) {
-            throw "Compilation of $Lib failed"
-        }
-    } elseif ($IsWindows -and (-not $SMAOnly)) {
-        log "Start building native Windows binaries"
+            if (-not (Test-Path $Lib)) {
+                throw "Compilation of $Lib failed"
+            }
+        } elseif ($IsWindows -and (-not $SMAOnly)) {
+            log "Start building native Windows binaries"
 
-        try {
-            Push-Location "$PSScriptRoot\src\powershell-native"
+            try {
+                Push-Location "$PSScriptRoot\src\powershell-native"
 
-            $NativeHostArch = "x64"
-            if ($script:Options.Runtime -match "-x86")
-            {
-                $NativeHostArch = "x86"
-            }
+                $NativeHostArch = "x64"
+                if ($script:Options.Runtime -match "-x86")
+                {
+                    $NativeHostArch = "x86"
+                }
 
-            # setup cmakeGenerator
-            if ($NativeHostArch -eq 'x86') {
-                $cmakeGenerator = 'Visual Studio 14 2015'
-            } else {
-                $cmakeGenerator = 'Visual Studio 14 2015 Win64'
-            }
+                # setup cmakeGenerator
+                if ($NativeHostArch -eq 'x86') {
+                    $cmakeGenerator = 'Visual Studio 14 2015'
+                } else {
+                    $cmakeGenerator = 'Visual Studio 14 2015 Win64'
+                }
 
-            # Compile native resources
-            $currentLocation = Get-Location
-            @("nativemsh/pwrshplugin") | % {
-                $nativeResourcesFolder = $_
-                Get-ChildItem $nativeResourcesFolder -Filter "*.mc" | % {
-                    $command = @"
-cmd.exe /C cd /d "$currentLocation" "&" "$($vcVarsPath)\vcvarsall.bat" "$NativeHostArch" "&" mc.exe -o -d -c -U "$($_.FullName)" -h "$nativeResourcesFolder" -r "$nativeResourcesFolder"
+                # Compile native resources
+                $currentLocation = Get-Location
+                @("nativemsh/pwrshplugin") | % {
+                    $nativeResourcesFolder = $_
+                    Get-ChildItem $nativeResourcesFolder -Filter "*.mc" | % {
+                        $command = @"
+    cmd.exe /C cd /d "$currentLocation" "&" "$($vcVarsPath)\vcvarsall.bat" "$NativeHostArch" "&" mc.exe -o -d -c -U "$($_.FullName)" -h "$nativeResourcesFolder" -r "$nativeResourcesFolder"
```

Comment:
I don't think this was meant to be indented. I doubt it'll break anything, but it looks accidental.